### DISCUSSION TITLE
Initial setup - Default values

### DISCUSSION THREAD

**Discussion by landerop** [deptagency/algomart#65] [General]
Hi team, I've been playing around with the platform and I had some issues while running `docker-compose up` 

```
algomart-api | [1636942003962] ERROR (197 on 1685b02f2ad8): the mnemonic contains a word that is not in the wordlist
algomart-api |     Error: the mnemonic contains a word that is not in the wordlist
algomart-api |         at seedFromMnemonic (/app/node_modules/algosdk/src/mnemonic/mnemonic.ts:106:42)
algomart-api |         at Object.mnemonicToSecretKey (/app/node_modules/algosdk/src/mnemonic/mnemonic.ts:147:16)
algomart-api |         at new AlgorandAdapter (/app/services/api/src/lib/algorand-adapter.ts:38:35)
algomart-api |         at Object.create (/app/services/api/src/shared/dependency-resolver.ts:89:7)
algomart-api |         at DependencyResolver.get (/app/services/api/src/shared/dependency-resolver.ts:74:23)
algomart-api |         at Object.create (/app/services/api/src/shared/dependency-resolver.ts:147:11)
algomart-api |         at DependencyResolver.get (/app/services/api/src/shared/dependency-resolver.ts:74:23)
algomart-api |         at Object.create (/app/services/api/src/shared/dependency-resolver.ts:138:11)
algomart-api |         at DependencyResolver.get (/app/services/api/src/shared/dependency-resolver.ts:74:23)
algomart-api |         at Object.create (/app/services/api/src/shared/dependency-resolver.ts:171:11)
```

So I was wondering if that has to do with the default values on .env file, what would be the default values for:
# API configuration
HOST=
PORT=
API_KEY=

# Algod configuration
ALGOD_SERVER=
ALGOD_PORT=
ALGOD_TOKEN=

FUNDING_MNEMONIC=

**Comment by afraser** [comment]
It looks like you still need to go through the initial setup. The error you're seeing is likely because you haven't generated a mnemonic in algorand sandbox see [services/api/README.md](https://github.com/deptagency/algomart/blob/main/services/api/README.md) for how to do this.

As for the other env vars, this is what I use:
```
HOST=
PORT=3001
API_KEY=<<make something up, you'll use this in the CMS setup outlined in apps/cms/README.md>>
ALGOD_SERVER=http://localhost
ALGOD_PORT=4001
ALGOD_TOKEN=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
```

@smonn Should we update the example env files to include sensible defaults for more of these variables?

**Reply by landerop** [reply]
Yeah that was exactly what was happening. I got that solved.

**Comment by smonn** [comment]
We can certainly add default values for these to help get you started. The Algod ones can default to the [sandbox](https://github.com/algorand/sandbox#usage) values. As @afraser pointed out, I'll defer to the [API README](https://github.com/deptagency/algomart/blob/main/services/api/README.md#creating-a-local-algorand-sandbox-account) for creating a new account, but we can add instructions there about funding it.

**Reply by landerop** [reply]
Yeah, that would be awesome. I got it sorted but if more instructions can be added, that's much better.

**Comment by djmbdv** [comment]
To GET /

HTTPError: Request failed with status code 401 Unauthorized

**Reply by landerop** [reply]
are you getting 401 when trying to connect Directus (CMS) with the API ? I'm getting this also. even when I've configured both tokens to be the same on API and CMS env files. Any idea @smonn ?

**Reply by djmbdv** [reply]
I getting 401 when I try to get  Homepage from the API

**Reply by smonn** [reply]
Does the CMS admin user have the same token set according to [these instructions](https://github.com/deptagency/algomart/tree/main/apps/cms#2-add-token-for-user)?

**Reply by landerop** [reply]
Somehow I'm having issues when trying to access Directus, with the credentials I put on config.local.json, it's not accepting them somehow. I'll do the whole process from scratch but the algorant sandbox and will get here

**Reply by smonn** [reply]
@landerop We've had issues where the admin user wasn't created correctly. What's the output from `npx directus count directus_users` (run within the `./apps/cms` directory)? If it's not 1 (or greater), you can generate a new admin user via these commands:

```bash
# create admin role, it'll output a UUID, use that in the next command
npx directus roles create --role admin --admin

# create admin user (replace <UUID>)
npx directus users create --email email@example.com --password t0ps3cret --role <UUID>
```

**Reply by landerop** [reply]
Let me try that and I'll get back. Thank you

**Comment by landerop** [comment]
First command gave me 1,  the default user created was admin@example.com so It didn't take my configuration inside config.local.json is that something you're aware of then?
I was able to  run the commands and create such user and got acces though :)

**Reply by afraser** [reply]
Hrm, I've run into issues with `npm run bootstrap` in CMS in the past but it was always related to the `npx directus roles create --role Admin --admin` command outputting a message in addition to the UUID so it wasn't getting properly interpolated into the second `npx directus users create` command (looks like we may still need to fix this so I'll open a PR).

I'm not sure how you'd get `admin@example.com` ...best guess is your config.local.json wasn't saved or was saved in the wrong place (needs to be in apps/cms).

We appreciate all the feedback on the setup process! It's not without it's bumps in the road (I've personally gone through the process several times and fallen flat on my face pretty much every time). I'm hoping to automate some of the CMS configuration steps that are particularly tricky (like setting the cms user token and allowing public read access to files)

**Reply by landerop** [reply]
Yes, somehow the bootstrap was complaining about some denied permissions so I had to manually grant permissions to cms role. I got that sorted, but I'm expecting to run a fresh installation from scratch to see if I face the same inconvenients again.
the config.local.json in fact was located in the correct location, weird.

**Reply by afraser** [reply]
I've got a fix for the cms bootstrap here FWIW: https://github.com/deptagency/algomart/pull/115/files

**Comment by landerop** [comment]
So seems I'm in the last stage of the configuration I'm trying to access: the Algorand Storefront, but I'm getting this:

```
Unhandled Runtime Error
FirebaseError: Firebase: Error (auth/invalid-api-key).
```
Definitely I haven't setup a Firebase project for this, still not clear on some things here (`NEXT_PUBLIC_FIREBASE_CONFIG` and `FIREBASE_SERVICE_ACCOUNT` ), for example some example values or link to official documentation where/ how to find that.

I see something that could be related here https://github.com/vercel/next.js/issues/13585

**Reply by landerop** [reply]
Nevermind,  I figured it out, `NEXT_PUBLIC_FIREBASE_CONFIG` is for the app config. `FIREBASE_SERVICE_ACCOUNT` is for the Firebase project itself. Still would be nice to clarify this for new users using the platform

**Comment by landerop** [comment]
Should I expect to be able to login to the storefront with the user I created on Directus?

**Reply by afraser** [reply]
No. You'll have to create a user in the algomart storefront UI.

 Also before you do this you should make sure you've followed the API setup instructions in services/api/README.md

**Reply by landerop** [reply]
Okay I'll do so. And yes, I've made sure I ran every step on api/README.md and cms/README.md.

**Comment by landerop** [comment]
Hey team, do you have any slack channel for the project? It would be good for a faster communication I think.

**Reply by smonn** [reply]
We have a channel (`#algomart`) on the official [Algorand Discord server](https://discord.gg/algorand).

**Comment by landerop** [comment]
Hey team, I'm stuck here, in the Storefront . I'm supposed to be allowed to get this but I've tried editing the `IMAGE_DOMAINS` variable. I've tried adding cms to my etc/hosts ..  any idea of what I'm doing wrong


![collections](https://user-images.githubusercontent.com/8417930/144007987-ffaf3d32-f754-4696-a959-d92dd30c8d9e.png)

**Reply by jakerainis** [reply]
Did you make the files publicly viewable in Directus? https://github.com/deptagency/algomart/tree/main/apps/cms#1-make-files-publicly-viewable
